### PR TITLE
fix(httpx): manually construct removed `URL.raw` property

### PR DESCRIPTION
**Pull request by Yun-Kim**
## Description
[As of `httpx==0.23.1`](https://github.com/encode/httpx/pull/2241), the `URL.raw` property was removed, which broke our `_url_to_str()` helper that uses that property to construct the raw URL.

However, the removed property was simply a combination of already existing (and still existing) properties in the `URL` object, which we can use to manually consturct the raw URL. I've version gated it such that nothing's changed for previous versions of `httpx`, but moving forward we'll manually construct the raw URL.

## Checklist
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by brettlangdon — APPROVED** [review]
one thought, but other lgtm

File: ddtrace/contrib/httpx/patch.py [hunk 1]
```diff
@@ -39,7 +41,14 @@ def _url_to_str(url):
     """
     Helper to convert the httpx.URL parts from bytes to a str
     """
-    scheme, host, port, raw_path = url.raw
+    # httpx==0.23.1 removed URL.raw, must construct it manually
+    if HTTPX_VERSION >= (0, 23, 1):
+        scheme = url.raw_scheme
+        host = url.raw_host
+        port = url.port
+        raw_path = url.raw_path
```
[the remark this answers is not in the document]
Comment (Yun-Kim) [reply ddtrace/contrib/httpx/patch.py:49]: httpx only added `URL.raw_scheme` and internally changed how `URL.raw_host` formats the URL  in versions 0.15 and 0.17 respectively IIRC, so it's not feasible to do this for all cases 😢

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Comment by brettlangdon** [comment]
@Mergifyio backport 1.6

**Comment by brettlangdon** [comment]
@Mergifyio refresh

**Comment by brettlangdon** [comment]
Fixes #4593

**Pull request merged**